### PR TITLE
Fixed EH and error reporting in restore_from

### PR DESCRIPTION
**Pull request by borisfom**
Signed-off-by: Boris Fomitchev <bfomitchev@nvidia.com>

# What does this PR do ?

Fixes 'lost cascading error info' due to constructor not being under try:.
As a result, my previously added code to save initial error was not executed. 
Also streamlined control flow in that block.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by borisfom** [comment]
Well, I realized the actual problem was fixed in main right before  ... so that fix only streamlines flow and improves formatting.

File: nemo/core/classes/common.py [hunk 1]
```diff
@@ -585,7 +571,7 @@ def to_config_file(self, path2yaml_file: str):
         """
         if hasattr(self, '_cfg'):
             self._cfg = maybe_update_config_version(self._cfg)
-            with open(path2yaml_file, 'w', encoding='utf-8') as fout:
+            with open(path2yaml_file, 'w') as fout:
```
Comment (titu1994) [inline nemo/core/classes/common.py:574]: This is required. It causes encoding errors across all Nemo models.

**Review by titu1994 — CHANGES_REQUESTED** [review]

**Comment by titu1994** [comment]
The other changes look great

File: nemo/core/classes/common.py (hunk 1 above)
Comment (borisfom) [reply nemo/core/classes/common.py]: Done.

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**